src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5268,7 +5268,7 @@ void MacroAssembler::char_array_compress(Register src, Register dst, Register le
 // aarch64_get_thread_helper() clobbers only r0, r1, and flags.
 //
 void MacroAssembler::get_thread(Register dst) {
-  RegSet saved_regs = RegSet::range(r0, r1) + lr - dst;
+  RegSet saved_regs = RegSet::range(r0, r1) + BSD_ONLY(RegSet::range(r2, r17)) + lr - dst;
```

Comment:
The comment needs to be updated, since on BSD we also seem to clobber r2,r17 ?

Reply:
Surely this should be 

```RegSet saved_regs = RegSet::range(r0, r1) BSD_ONLY(+ RegSet::range(r2, r17)) + lr - dst;```

Shouldn't it?

Reply:
Interesting, I wonder why it has built successfully on Linux. I'm going to fix this under as JDK-8261072